src/kernels/kernelDependencyService.ts

```diff
@@ -83,14 +89,25 @@ export class KernelDependencyService implements IKernelDependencyService {
         }
 
         // Cache the install run
-        const cancelTokenSource = new CancellationTokenSource();
         let promise = this.installPromises.get(kernelConnection.interpreter.path);
+        let cancelTokenSource: CancellationTokenSource | undefined;
         if (!promise) {
+            const cancelTokenSource = new CancellationTokenSource();
+            const disposable = token.onCancellationRequested(() => {
```

Comment:
This was a bug, basically we stop execution, but this part continues, as we created a whole new cancellation token and ignored the one that was passed into this.